build error using TS 5.6

### What version of Effect is running?

3.10.4

### What steps can reproduce the bug?

have dependency of effect ts 3.10.4 and using TS 5.6

### What is the expected behavior?

build correctly

### What do you see instead?

../../node_modules/.pnpm/effect@3.10.4/node_modules/effect/dist/dts/Utils.d.ts:69:22 - error TS2420: Class 'SingleShotGen<T, A>' incorrectly implements interface 'Generator<T, A, any>'.
  Type 'SingleShotGen<T, A>' is missing the following properties from type 'Generator<T, A, any>': map, filter, take, drop, and 9 more.

69 export declare class SingleShotGen<T, A> implements Generator<T, A> {
                        ~~~~~~~~~~~~~


Found 1 error in ../../node_modules/.pnpm/effect@3.10.4/node_modules/effect/dist/dts/Utils.d.ts:69

### Additional information

if build with TS 5.5, it's successful.

--- a/.changeset/polite-crews-march.md
+++ b/.changeset/polite-crews-march.md
@@ -1,0 +1,5 @@
+---
+"effect": patch
+---
+
+extend `IterableIterator` instead of `Generator` in `SingleShotGen`

--- a/packages/effect/src/Utils.ts
+++ b/packages/effect/src/Utils.ts
@@ -106,7 +106,7 @@ export class GenKindImpl<F extends TypeLambda, R, O, E, A> implements GenKind<F,
  * @category constructors
  * @since 2.0.0
  */
-export class SingleShotGen<T, A> implements Generator<T, A> {
+export class SingleShotGen<T, A> implements IterableIterator<T, A> {
   private called = false
 
   constructor(readonly self: T) {}

--- a/packages/effect/src/Utils.ts
+++ b/packages/effect/src/Utils.ts
@@ -40,7 +40,7 @@ export type GenKindTypeId = typeof GenKindTypeId
 export interface GenKind<F extends TypeLambda, R, O, E, A> extends Variance<F, R, O, E> {
   readonly value: Kind<F, R, O, E, A>
 
-  [Symbol.iterator](): Generator<GenKind<F, R, O, E, A>, A>
+  [Symbol.iterator](): IterableIterator<GenKind<F, R, O, E, A>, A>
 }
 
 /**
@@ -97,7 +97,7 @@ export class GenKindImpl<F extends TypeLambda, R, O, E, A> implements GenKind<F,
   /**
    * @since 2.0.0
    */
-  [Symbol.iterator](): Generator<GenKind<F, R, O, E, A>, A> {
+  [Symbol.iterator](): IterableIterator<GenKind<F, R, O, E, A>, A> {
     return new SingleShotGen<GenKind<F, R, O, E, A>, A>(this as any)
   }
 }
@@ -147,7 +147,7 @@ export class SingleShotGen<T, A> implements IterableIterator<T, A> {
   /**
    * @since 2.0.0
    */
-  [Symbol.iterator](): Generator<T, A> {
+  [Symbol.iterator](): IterableIterator<T, A> {
     return new SingleShotGen<T, A>(this.self)
   }
 }

